src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -360,13 +360,65 @@ private BoundStatement BindThrow(ThrowStatementSyntax node, DiagnosticBag diagno
             return new BoundThrowStatement(node, boundExpr, hasErrors);
         }
 
-        private static BoundStatement BindEmpty(EmptyStatementSyntax node)
+        private static bool IsEmbeddedStatementParent(SyntaxNode node)
```

Comment:
>IsEmbeddedStatementParent [](start = 28, length = 25)

It feels like we don't need a helper like this, it will be sufficient to perform necessary checks in BindPossibleEmbeddedStatement. In there we can be sure that statements of the interest (an empty statement, declarations, labeled statement) are embedded statements. At the same time, no embedded statement should bypass that method.